regular_shapes.scad

```diff
@@ -41,9 +41,12 @@ module pentagon(radius)
   reg_polygon(5,radius);
 }
 
-module hexagon(radius)
+module hexagon(radius, diameter, across_flats)
 {
-  reg_polygon(6,radius);
+  r = radius;
+  r = diameter ? diameter/2 : radius;
+  r = across_flats ? across_flats/2/cos(30) : radius;
```

Comment:
You can merge these statements into one:
```scad
r = across_flats ? across_flats/2/cos(30) : diameter ? diameter/2 : radius;
```

That also solves the issue where the third statement overrides the diameter in the case of `hexagon(diameter=10)`

Reply:
Good catch. Thanks a lot for looking at this.